### PR TITLE
Fixed typos and grammar in lib/Syntax/README.md

### PR DESCRIPTION
**Pull request by NevinBR**
Minor spelling and grammar corrections in the lib/Syntax readme.

### CODE REVIEW

File: lib/Syntax/README.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ StructDeclSyntaxBuilder Builder;
 // We previously parsed a struct keyword, let's tell the builder to use it.
 Builder.useStructKeyword(StructKeyword);
 
-// Hm, we didn't see an identifier, but saw a left brace. Let's keep going.
```
Comment (xwu) [inline lib/Syntax/README.md]: This is fine as written.

File: lib/Syntax/README.md [hunk 2]
```diff
@@ -163,8 +163,8 @@ struct YourStruct {}
 At any point in the building process, you can call `build()` and get a
 reasonably formed Syntax node (i.e. with no raw `nullptr`s) using what you've
 provided to the builder so far. Anything that you haven't supplied is marked as
-*missing*. This is essentially what the parser does so, looking forward to
-future adoption, the builders are designed with the parser in mind, with the
```
Comment (xwu) [inline lib/Syntax/README.md]: The punctuation here is more correct as written than in the revised version.

Comment (NevinBR) [reply lib/Syntax/README.md]: I strongly disagree. There needs to be a comma between “does” and “so”.

After that it is more stylistic, and in my view reads better as revised.

File: lib/Syntax/README.md (hunk 1 above)
Comment (NevinBR) [reply lib/Syntax/README.md]: The original uses a non-standard variant spelling.

File: lib/Syntax/README.md (hunk 2 above)
Comment (harlanhaskins) [reply lib/Syntax/README.md]: How about: "This is essentially what the parser does, so—looking forward to future adoption—the builders are designed with the parser in mind, with the

File: lib/Syntax/README.md (hunk 1 above)
Comment (harlanhaskins) [reply lib/Syntax/README.md]: I can take this or leave it. 'Hm' is just as correct as 'Hmm'.

**Comment by harlanhaskins** [comment]
Thanks for looking thoroughly thorough this document!

File: lib/Syntax/README.md (hunk 1 above)
Comment (NevinBR) [reply lib/Syntax/README.md]: Okay I’m outvoted, we’ll keep it as “Hm”.

File: lib/Syntax/README.md (hunk 2 above)
Comment (NevinBR) [reply lib/Syntax/README.md]: Good thinking, Harlan, I’ll make it use em-dashes.

Comment (xwu) [reply lib/Syntax/README.md]: Commas are more correct; em dashes are used to offset [shifts in thought](http://theweek.com/articles/460264/youre-using-that-dash-wrong):

> It's often a bad idea to use it where you would use a comma — unless the comma is at a clause boundary and you want to signify a greater break.

Comment (NevinBR) [reply lib/Syntax/README.md]: Em dashes are used for [other things](https://en.wikipedia.org/wiki/Dash#Em_dash) as well, and would be perfectly correct here. However, I have reverted them to commas, and placed a semicolon between “does” and “so”, thus breaking up the run-on sentence after its first independent clause.

**Comment by CodaFi** [comment]
Will squash this down manually.

Thanks!

@swift-ci please smoke test

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**